pandas/core/generic.py

```diff
@@ -4272,8 +4272,14 @@ def _align_series(self, other, join='outer', axis=None, level=None,
 
         Parameters
         ----------
-        cond : boolean %(klass)s or array
-        other : scalar or %(klass)s
+        cond : boolean %(klass)s, array or callable
+            If cond is callable, it is computed on the %(klass)s and
+            should return boolean %(klass)s or array.
+            The callable must not change input %(klass)s.
```

Comment:
Perhaps note whether or not pandas checks if the type has changed (I haven't gotten there yet to see).